pkg/flowaggregator/clickhouseclient/clickhouseclient.go

```diff
@@ -0,0 +1,511 @@
+	}
+	if egressNPType, _, ok := record.GetInfoElementWithValue("egressNetworkPolicyType"); ok {
+		chFlowRow.egressNetworkPolicyType = egressNPType.GetUnsigned8Value()
+	}
+	if egressNPRuleAction, _, ok := record.GetInfoElementWithValue("egressNetworkPolicyRuleAction"); ok {
+		chFlowRow.egressNetworkPolicyRuleAction = egressNPRuleAction.GetUnsigned8Value()
+	}
+	if tcpState, _, ok := record.GetInfoElementWithValue("tcpState"); ok {
+		chFlowRow.tcpState = tcpState.GetStringValue()
+	}
+	if flowType, _, ok := record.GetInfoElementWithValue("flowType"); ok {
+		chFlowRow.flowType = flowType.GetUnsigned8Value()
+	}
+	if sourcePodLabels, _, ok := record.GetInfoElementWithValue("sourcePodLabels"); ok {
+		chFlowRow.sourcePodLabels = sourcePodLabels.GetStringValue()
+	}
+	if destinationPodLabels, _, ok := record.GetInfoElementWithValue("destinationPodLabels"); ok {
+		chFlowRow.destinationPodLabels = destinationPodLabels.GetStringValue()
+	}
+	if throughput, _, ok := record.GetInfoElementWithValue("throughput"); ok {
+		chFlowRow.throughput = throughput.GetUnsigned64Value()
+	}
+	if reverseThroughput, _, ok := record.GetInfoElementWithValue("reverseThroughput"); ok {
+		chFlowRow.reverseThroughput = reverseThroughput.GetUnsigned64Value()
+	}
+	if throughputFromSrcNode, _, ok := record.GetInfoElementWithValue("throughputFromSourceNode"); ok {
+		chFlowRow.throughputFromSourceNode = throughputFromSrcNode.GetUnsigned64Value()
+	}
+	if throughputFromDstNode, _, ok := record.GetInfoElementWithValue("throughputFromDestinationNode"); ok {
+		chFlowRow.throughputFromDestinationNode = throughputFromDstNode.GetUnsigned64Value()
+	}
+	if revTputFromSrcNode, _, ok := record.GetInfoElementWithValue("reverseThroughputFromSourceNode"); ok {
+		chFlowRow.reverseThroughputFromSourceNode = revTputFromSrcNode.GetUnsigned64Value()
+	}
+	if revTputFromDstNode, _, ok := record.GetInfoElementWithValue("reverseThroughputFromDestinationNode"); ok {
+		chFlowRow.reverseThroughputFromDestinationNode = revTputFromDstNode.GetUnsigned64Value()
+	}
+	return &chFlowRow
+}
+
+func (ch *ClickHouseExportProcess) flowRecordPeriodicCommit() {
+	commitTicker := time.NewTicker(commitInterval)
+	logTicker := time.NewTicker(time.Minute)
+	committedRec := 0
+	for {
+		select {
+		case <-ch.stopChan:
+			commitTicker.Stop()
+			logTicker.Stop()
+			return
+		case <-commitTicker.C:
+			committed, err := ch.batchCommitAll()
+			if err == nil {
+				committedRec += committed
+			}
+		case <-logTicker.C:
+			klog.V(4).InfoS("Total number of records committed to DB", "count", committedRec)
+			committedRec = 0
+		}
+	}
+}
+
+func (ch *ClickHouseExportProcess) batchCommitAll() (int, error) {
+	currSize := ch.deque.Len()
+	if currSize == 0 {
+		return 0, nil
+	}
+
+	var stmt *sql.Stmt
+
+	// start new connection
+	tx, err := ch.db.Begin()
+	if err == nil {
+		stmt, err = tx.Prepare(insertQuery)
+	}
+	if err != nil {
+		klog.ErrorS(err, "Error when preparing insert statement")
+		_ = tx.Rollback()
```

Comment:
Sorry for late comment on the previous commit. Curious, we don't need to deal with tx.Rollback fails situation?

Reply:
errors thrown by Rollback is partially implementation specific per sql interface. For CH the only possible err are:
- net/net.go conn.Close err
- Rollback called not within Transaction

1) does not need FA to handle and 2) is N/A. The err was also ignored in CH examples.